src/shared/components/TimePicker/TimePicker.tsx

```diff
@@ -21,58 +24,99 @@ export const TimePicker = <T extends FieldValues>({
   minTime,
   maxTime,
   onCustomChange,
+  defaultTime = DEFAULT_TIME,
   'data-testid': dataTestid,
   placeholder,
   inputSx = {},
-}: TimePickerProps<T>) => (
-  <Controller
-    control={control}
-    name={name}
-    render={({ field: { onChange, value }, fieldState: { error } }) => {
-      const selected = value ? parse(value, DateFormats.Time, new Date()) : value;
-      const handleChange = (date: Date) => {
-        const newTime = dateFnsFormat(date, DateFormats.Time);
-        onCustomChange?.(newTime);
-        onChange(newTime);
-      };
+}: TimePickerProps<T>) => {
+  const [inputValue, setInputValue] = useState<string>('');
+
+  const updateInputValue = (value: string, onChange: (value: string) => void) => {
+    onCustomChange?.(value);
+    setInputValue(value);
+    onChange(value);
+  };
+
+  const handleInputValue = (onChange: (value: string) => void) => {
+    const cleanedInput = cleanInput(inputValue);
+    const validatedInput = validateInput(defaultTime, cleanedInput);
+    const formattedInput = validatedInput.length === 4 ? formatInput(validatedInput) : defaultTime;
```

Comment:
If I'm typing '12' and press enter, is it expected to output '12:00'? Maybe it's also good idea to take such cases into account.